### PR TITLE
chore: Add documentations for superfluid module

### PR DESCRIPTION
**Pull request by mattverse**
sub component of: #1955

## What is the purpose of the change

This PR adds proto message documentation and additional documentation for msg_server and code within `x/superfluid/keeper`. 

This PR does not include ANY change or refactoring for any of the logic within `x/superfluid`

## Brief Changelog
- Add documentation for `x/superfluid`

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? additional documentations

### CODE REVIEW

File: proto/osmosis/superfluid/params.proto [hunk 1]
```diff
@@ -8,7 +8,7 @@ option go_package = "github.com/osmosis-labs/osmosis/v7/x/superfluid/types";
 
 // Params holds parameters for the superfluid module
 message Params {
-  // the risk_factor is to be cut on OSMO equivalent value of lp tokens for
+  // MinimumRiskFactor is to be cut on OSMO equivalent value of lp tokens for
```
[the remark this answers is not in the document]
Comment (mattverse) [reply proto/osmosis/superfluid/params.proto:11]: Elabortaion added!

File: proto/osmosis/superfluid/genesis.proto [hunk 2]
```diff
@@ -10,10 +10,16 @@ option go_package = "github.com/osmosis-labs/osmosis/v7/x/superfluid/types";
 // GenesisState defines the module's genesis state.
 message GenesisState {
   Params params = 1 [ (gogoproto.nullable) = false ];
+  // SuperfluidAssets defines the registered superfluid assets that has been
```
[the remark this answers is not in the document]
Comment (mattverse) [reply proto/osmosis/superfluid/genesis.proto:13]: Fixed all changes to proto cases!

File: x/superfluid/keeper/msg_server.go [hunk 3]
```diff
@@ -29,6 +29,18 @@ func NewMsgServerImpl(keeper *Keeper) types.MsgServer {
...
+// This requires the lock to have locked tokens that has been already registeed as a superfluid asset via governance.
+// The pre-requisites for a lock to be able to be eligible for superfluid delegation are
+// - assets in the lock should be a superfluid registered asset
+// - lock should only have a single asset
+// - lock should not be unlocking
+// - lock should not have a different superfluid staking position
+// - lock duration should be greater or equal to the staking.Unbonding time
+// Note that the amount of delegation is not equal to the equivalent amount of osmo within the lock.
+// Instead, we use the osmo equivalent multiplier stored in the latest epoch, calculate how much
+// osmo equivalent is in lock, and use the risk adjusted osmo value.
```
[the remark this answers is not in the document]
Comment (mattverse) [reply x/superfluid/keeper/msg_server.go:42]: Addressed!

**Review by p0mvn — APPROVED** [review]

File: proto/osmosis/superfluid/genesis.proto [hunk 4]
```diff
@@ -10,10 +10,16 @@ option go_package = "github.com/osmosis-labs/osmosis/v7/x/superfluid/types";
 // GenesisState defines the module's genesis state.
 message GenesisState {
   Params params = 1 [ (gogoproto.nullable) = false ];
+  // superfluid_assets defines the registered superfluid assets that have been
+  // registered via governance.
   repeated SuperfluidAsset superfluid_assets = 2
       [ (gogoproto.nullable) = false ];
+  // osmo_equivalent_multipliers is the records of osmo equivilent amount of
+  // each superfluid registered pool, updated every epoch.
```
Comment (ValarDragon) [inline proto/osmosis/superfluid/genesis.proto:18]:
```suggestion
  // osmo_equivalent_multipliers is the records of osmo equivalent amount of
  // each superfluid registered pool, updated every epoch.
```

File: proto/osmosis/superfluid/params.proto [hunk 5]
```diff
@@ -8,8 +8,9 @@ option go_package = "github.com/osmosis-labs/osmosis/v7/x/superfluid/types";
 
 // Params holds parameters for the superfluid module
 message Params {
-  // the risk_factor is to be cut on OSMO equivalent value of lp tokens for
-  // superfluid staking, default: 5%
+  // minimum_risk_factor is to be cut on OSMO equivalent value of lp tokens for
+  // superfluid staking, default: 5%. The minimum risk facotr works as a
+  // parameter to better incentive vanilla staking.
```
Comment (ValarDragon) [inline proto/osmosis/superfluid/params.proto:13]:
```suggestion
  // superfluid staking, default: 5%. The minimum risk factor works 
  // to counter-balance the staked amount on chain's exposure to various asset volatilities,
  // and have base staking be 'resistant' to volatility.
```

**Review by ValarDragon — APPROVED** [review]
Nice job! Good to merge imo once typo fixes are in

**Comment by mattverse** [comment]
L forgot to push commit's with @ValarDragon 's recent comments before merging this, apologies for the confusion, creating a secondary PRs with the fixes :(

**Pull request merged**